neurots/generate/tree.py

```diff
@@ -248,8 +248,23 @@ def next_point(self):
         for section_grower in ordered_sections:
             # the current section_grower is generated
             # In here the stop criterion can be modified accordingly
+
+            if (
+                section_grower.process == "major" and "cut_major_length" in self.params
```

Comment:
I would give a more informative name to this parameter: 
"early_terminate_major"
"major_termination_length"

Reply:
ok, for `major_termination_length'!